pkg/resource/deploy/step.go

```diff
@@ -1011,12 +1027,22 @@ func (s *RefreshStep) Apply(preview bool) (resource.Status, StepCompleteFunc, er
 			s.old.Parent, s.old.Protect, s.old.External, s.old.Dependencies, initErrors, s.old.Provider,
 			s.old.PropertyDependencies, s.old.PendingReplacement, s.old.AdditionalSecretOutputs, s.old.Aliases,
 			&s.old.CustomTimeouts, s.old.ImportID, s.old.RetainOnDelete, s.old.DeletedWith, s.old.Created, s.old.Modified,
-			s.old.SourcePosition,
+			s.old.SourcePosition, s.old.IgnoreChanges,
 		)
 		var inputsChange, outputsChange bool
 		if s.old != nil {
-			inputsChange = !refreshed.Inputs.DeepEquals(s.old.Inputs)
-			outputsChange = !refreshed.Outputs.DeepEquals(s.old.Outputs)
+			// Prior to us introducing refresh diffs against desired state, we only
+			// diffed outputs. For now, we support reverting to this behaviour for
+			// users that rely on it. We only need to make this check here as all
+			// downstream consumers will behave correctly based on whether s.diff is
+			// set or not.
```

Comment:
```suggestion
			// users that rely on it.
```

This is the second place where we do this check, so "We only need to make this check here" isn't quite correct.